lib/merchant_repository.rb

```diff
@@ -76,4 +76,7 @@ def find_invoices(id)
     sales_engine.find_invoices_by_merchant_id(id)
   end
 
+  def most_revenue(x)
+    merchants.sort_by { |merchant| merchant.revenue }.reverse[0..(x-1)]
```

Comment:
Surrounding space missing for operator '-'.<br>Pass `&:revenue` as an argument to `sort_by` instead of a block.